src/blob_gc_picker.cc

```diff
@@ -72,20 +64,62 @@ std::unique_ptr<BlobGC> BasicBlobGCPicker::PickBlobGC(
                          next_gc_size);
           break;
         }
+      }
+    }
+  }
+   // files with larger discardable size Free Space first
+  std::sort(gc_scores.begin(), gc_scores.end(),
+            [](const GCScore& first, const GCScore& second) {
+              return first.fs_score > second.fs_score;
+            });
+
+  std::vector<BlobFileMeta*> fs_blob_files;
+  uint64_t next_fs_size = 0;
+  uint64_t fs_batch_size = 0;
+  stop_picking = false;
+  for (auto& fs_score: gc_scores) {
+    auto blob_file = blob_storage->FindFile(fs_score.file_number).lock();
+    if (!blob_file ||
+        blob_file->file_state() == BlobFileMeta::FileState::kBeingGC ||
+        !CheckBlobFile(blob_file.get())) {
+      continue;
+    }
+
+    if (!stop_picking) {
+      fs_blob_files.push_back(blob_file.get());
+      fs_batch_size += blob_file->file_size();
+      if (fs_batch_size >= cf_options_.max_fs_batch_size) {
+        stop_picking = true;
+      }
+    } else {
+      if (maybe_continue_next_time) {
+        break;
+      }
+      if (blob_file->discardable_size() >= cf_options_.free_space_threshold) {
+        next_fs_size += blob_file->file_size();
+        if (next_fs_size > cf_options_.min_fs_batch_size) {
+          maybe_continue_next_time = true;
+          ROCKS_LOG_INFO(db_options_.info_log,
+                         "remain more than %" PRIu64
+                         " bytes to be gc and trigger after this gc",
+                         next_fs_size);
+          break;
+        }
       } else {
         break;
       }
     }
   }
+
   ROCKS_LOG_DEBUG(db_options_.info_log,
                   "got batch size %" PRIu64 ", estimate output %" PRIu64
                   " bytes",
-                  batch_size, estimate_output_size);
-  if (blob_files.empty() || batch_size < cf_options_.min_gc_batch_size)
+                  gc_batch_size, estimate_output_size);
+  if ((gc_blob_files.empty() || gc_batch_size < cf_options_.min_gc_batch_size) && (fs_blob_files.empty() || fs_batch_size < cf_options_.min_fs_batch_size))
```

Comment:
What if gc_blob_files is empty and fs_blob_files isn't?

Reply:
then this expression will be false, and we will  return a BlobGC, and in BlobGCJob::Run we will just run the fs job